kasa/klapprotocol.py

```diff
@@ -0,0 +1,500 @@
+            except TimeoutError as ex:
+                raise SmartDeviceException(
+                    f"Unable to connect to the device, timed out: {self.host}: {ex}"
+                )
+            except AuthenticationException as auex:
+                _LOGGER.debug("Unable to authenticate with %s, not retrying", self.host)
+                raise auex
+            except Exception as ex:
+                if retry >= retry_count:
+                    _LOGGER.debug("Giving up on %s after %s retries", self.host, retry)
+                    raise SmartDeviceException(
+                        f"Unable to connect to the device: {self.host}: {ex}"
+                    )
+                continue
+
+        # make mypy happy, this should never be reached..
+        raise SmartDeviceException("Query reached somehow to unreachable")
+
+    async def _execute_query(self, request: str, retry_count: int) -> Dict:
+        timeout = aiohttp.ClientTimeout(total=self.timeout)
+
+        async with aiohttp.ClientSession(
+            cookie_jar=self.jar, timeout=timeout
+        ) as session:
+            if not self.handshake_done:
+                try:
+                    await self.perform_handshake(session)
+
+                except AuthenticationException as auex:
+                    _LOGGER.debug(
+                        f"Unable to complete handshake for device {self.host}, authentication failed"
+                    )
+                    raise auex
+
+            # Check for mypy
+            if self.encryption_session is not None:
+                payload, seq = self.encryption_session.encrypt(request.encode())
+
+            url = f"http://{self.host}/app/request"
+
+            cookie = self.jar.filter_cookies(url).get(self.TP_SESSION_COOKIE_NAME)
+
+            response_status, response_data = await self.session_post(
+                session, url, params={"seq": seq}, data=payload
+            )
+
+            if response_status != 200:
+                _LOGGER.error(
+                    f"Query failed after succesful authentication at {datetime.datetime.now()}.  Host is {self.host}, Session cookie is {cookie}, Retry count is {retry_count}, Sequence is {seq}, Response status is {response_status}, Request was {request}"
+                )
+                # If we failed with a security error, force a new handshake next time.
+                if response_status == 403:
+                    self.handshake_done = False
+                    self.authentication_failed = True
+                    raise AuthenticationException(
+                        f"Got a security error from {self.host} after handshake completed {self.discovery_data}"
+                    )
+                else:
+                    raise SmartDeviceException(
+                        "Device %s responded with %d to request with seq %d"
+                        % (self.host, response_status, seq)
+                    )
+            else:
+                _LOGGER.debug(
```

Comment:
https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-not-lazy.html